docs/cryptodoc/src/05_09_kyber.rst

```diff
@@ -141,161 +302,115 @@ Based on these functions the key generation process follows **Algorithms 4 and 7
+They use the K-PKE encapsulation algorithm (Algorithm 14 of [FIPS-203]_)
+implemented in
+:srcref:`[src/lib/pubkey/kyber]/kyber/kyber_common/kyber_keys.cpp:55|Kyber_PublicKeyInternal::indcpa_encrypt`.
+In combination, Botan does the following:
 
-Encryption works as follows, realizing **Algorithm 5** of [Kyber-R3]_:
-
-.. admonition:: Kyber_KEM_Cryptor::indcpa_enc()
+.. admonition:: ML_KEM_Encryptor::encapsulate
 
    **Input:**
 
-   - ``pk = (pk_t, seed)``: public key
-   - ``m``: message
-   - ``coins``: randomness (input :math:`r` in Alg. 5 [Kyber-R3]_)
+   -  ``rng``: random number generator
+   -  ``pk = (t, rho)``: public key
 
    **Output:**
 
-   - ``c``: ciphertext bytes
+   -  ``K``: shared secret key
+   -  ``c``: ciphertext
 
    **Steps:**
 
-   1. ``at = PolynomialMatrix::generate(seed, true, mode)`` (``true`` means "transposed") (L. 3-8, Alg. 5 [Kyber-R3]_)
-   2. ``sp = PolynomialVector::getnoise_eta1(coins, 0, mode)`` (performs ``k`` invocations of ``Polynomial::getnoise_eta1``, one for each component of ``sp``; L. 9-12, Alg. 5 [Kyber-R3]_)
-   3. ``ep = PolynomialVector::getnoise_eta2(coins, k, mode)`` (performs ``k`` invocations of ``Polynomial::getnoise_eta2``, one for each component of ``ep``; L. 13-16, Alg. 5 [Kyber-R3]_)
-   4. ``epp = Polynomial::getnoise_eta2(coins, 2*k, mode)`` (L. 17, Alg. 5 [Kyber-R3]_)
-   5. ``sp.ntt()`` (L. 18, Alg. 5 [Kyber-R3]_)
-   6. ``bp = (at * sp).invntt() + ep`` (L. 19, Alg. 5 [Kyber-R3]_)
-   7. ``v = (pk_t * sp).invntt() + epp + Polynomial::from_message(m)`` (L. 20, Alg. 5 [Kyber-R3]_)
-   8. ``c = Ciphertext(bp, v, mode).to_bytes()`` (L. 21-23, Alg. 5 [Kyber-R3]_)
-
-   **Notes:**
-
-   - The member function ``Polynomial::getnoise_eta1(seed, nonce, mode)`` uses ``PRF`` on the seed with incremented nonce values to call ``Polynomial::getnoise_cbd2`` or ``Polynomial::getnoise_cbd3`` depending on ``eta1``.
-   - The member function ``Polynomial::getnoise_eta2(seed, nonce, mode)`` uses ``PRF`` on the seed with incremented nonce values to call ``Polynomial::getnoise_cbd2`` (as for all parameter sets ``eta2 = 2``).
-
-.. _pubkey/kyber/ccakem_enc:
-
-Kyber.CCAKEM
-^^^^^^^^^^^^
-
-Encapsulation works as follows, realizing **Algorithm 8** of [Kyber-R3]_:
-
-.. admonition:: Kyber_KEM_Encryptor::raw_kem_encrypt()
+   1. Generate a random message ``m`` using ``rng``
+   2. ``(K, r) = G(m || H(pk))``
+   3. K-PKE encrypt ``m`` using ``r`` to obtain ciphertext ``c``
 
-   **Input:**
-
-   - ``pk = (pk_t, seed)``: public key
-   - ``out_encapsulated_key``: ciphertext of shared key (to be overwritten)
-   - ``out_shared_key``: plaintext shared key (to be overwritten)
-   - ``rng``: random number generator
-
-   **Output:**
-
-   -  Overwritten ``out_encapsulated_key``, ``out_shared_key``
-
-   **Steps:**
+      4. Sample transposed matrix ``At`` from ``rho`` using ``sample_matrix``
+      5. Initialize a ``PolynomialSampler`` ``ps`` with ``sigma``
```

Comment:
```suggestion
      5. Initialize a ``PolynomialSampler`` ``ps`` with ``r``
```

Reply:
Good catch! You exposed my copy-paste fingers ;)